modules/caffeine/src/main/scala/scalacache/caffeine/CaffeineCache.scala

```diff
@@ -65,17 +60,17 @@ object CaffeineCache {
 
   /** Create a new Caffeine cache.
     */
-  def apply[F[_]: Sync: Clock, V](implicit config: CacheConfig): F[CaffeineCache[F, V]] =
-    Sync[F].delay(Caffeine.newBuilder().build[String, Entry[V]]()).map(apply(_))
+  def apply[F[_]: Sync: Clock, K <: AnyRef, V]: F[CaffeineCache[F, K, V]] =
```

Comment:
It looks like Caffeine requires the K and V to both extend `Object` (from Java). This wasn't a problem in the past because we were always using String keys. Weirdly, this also only seems to be an issue on Scala 2.12 as it compiles fine for 2.13 and 3. I had to add `<: AnyRef` just for 2.12, but I am not really sure of an alternative. @kubukoz Do you happen to have any thoughts on this?

Reply:
Added an issue #603 to discuss this further. Merging this PR so we can continue on with other things in the meantime.